Packet++/header/NdpLayer.h

```diff
@@ -0,0 +1,329 @@
+
+	OsiModelLayer getOsiModelLayer() const { return OsiModelNetworkLayer; }
+
+  private:
+	void setNeighborAdvertisementHeaderFields(const IPv6Address &targetIP, bool byRouter, bool unicastResponse,
+											  bool override);
+};
+
+/**
+ * @class NDPNeighborSolicitationLayer
+ * Represents an NDP Neighbor Solicitation protocol layer
+ */
+class NDPNeighborSolicitationLayer : public NDPLayerBase
+{
+  public:
+
+	/**
+	 * @struct ndpneighborsolicitationhdr
+	 * Represents neighbor solicitation message format
+	 */
+#pragma pack(push, 1)
+	struct ndpneighborsolicitationhdr
+	{
+		/** Reserved */
+		uint32_t reserved;
+		/** Target address - Target address of solicitation message */
+		uint8_t targetIP[16];
+	};
+#pragma pack(pop)
+
+	/**
+	 * A constructor that creates the layer from an existing packet raw data
+	 * @param[in] data A pointer to the raw data
+	 * @param[in] dataLen Size of the data in bytes
+	 * @param[in] prevLayer A pointer to the previous layer
+	 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+	 */
+	NDPNeighborSolicitationLayer(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet)
+		: NDPLayerBase(data, dataLen, prevLayer, packet)
+	{
+		m_Protocol = NDPNeighborSolicitation;
+	}
+
+	/**
+	 * A constructor for a new NDPNeighborSolicitationLayer object
+	 * @param[in] targetIP Target IP address for which the solicitation shall be created
+	 */
+	NDPNeighborSolicitationLayer(const IPv6Address &targetIP);
+
+	/**
+	 * A constructor for a new NDPNeighborSolicitationLayer object
+	 * @param[in] targetIP Target IP address for which the solicitation shall be created
+	 * @param[in] srcMac Mac address which shall be put in the linklayer option
+	 */
+	NDPNeighborSolicitationLayer(const IPv6Address &targetIP, const MacAddress &srcMac);
+
+	virtual ~NDPNeighborSolicitationLayer() {}
+
+	/**
+	 * @return A pointer to the @ref ndpneighborsolicitationhdr
+	 */
+	ndpneighborsolicitationhdr *getNdpHeader() const { return (ndpneighborsolicitationhdr *)m_Data; }
+
+	/**
+	 * @return Get the IP address specified as the target IP address in the solicitation message
+	 */
+	IPv6Address getTargetIP() const
+	{
+		return IPv6Address(getNdpHeader()->targetIP);
+	};
+
+	/**
+	 * Checks if the layer has a link layer address option set
+	 * @return true if link layer address option is available
+	 * @return false if not
+	 */
+	bool hasLinkLayerAddress() const;
+
+	/**
+	 * Get the Link Layer Address
+	 * @return Mac address which is specified in the link layer address option
+	 */
+	MacAddress getLinkLayerAddress() const;
+
+	/**
+	 * @return The length of the NDPNeighborSolicitation header.
+	 */
+	size_t getHeaderLen() const
+	{
+		return hasLinkLayerAddress() ? sizeof(ndpneighborsolicitationhdr) + sizeof(ndpoptionlinklayer)
+									 : sizeof(ndpneighborsolicitationhdr);
+	}
+
+	/**
+	 * This class is currently only used for parsing - so no implementation of this method
+	 */

```

Comment:
I think this comment is not up to date anymore.. I guess in the case of this layer nothing needs to be done here, even for the crafting usecase?